crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1846,7 +1852,13 @@ impl<'db> TypeInferenceBuilder<'db> {
 
                 unpacked.get(name_ast_id).unwrap_or(Type::Unknown)
             }
-            TargetKind::Name => value_ty,
+            TargetKind::Name => {
+                if self.file().is_stub(self.db().upcast()) && value.is_ellipsis_literal_expr() {
+                    Type::Any
```

Comment:
We should use `Type::Unknown` rather than `Type::Any` here. In our model, `Type::Any` is reserved for explicit uses of `typing.Any` as an annotation. Any dynamic type arising from lack of an annotation uses `Type::Unknown`.

Also we should add a test for this case (un-annotated assignment in a stub file, with `...` as RHS.)